packages/app/src/cli/services/dev/ui/components/DevSessionUI.tsx

```diff
@@ -17,40 +17,49 @@ import {openURL} from '@shopify/cli-kit/node/system'
 import figures from '@shopify/cli-kit/node/figures'
 import {isUnitTest} from '@shopify/cli-kit/node/context/local'
 import {treeKill} from '@shopify/cli-kit/node/tree-kill'
+import {isTruthy} from '@shopify/cli-kit/node/context/utilities'
 import {Writable} from 'stream'
 
 interface DevSesionUIProps {
   processes: OutputProcess[]
   abortController: AbortController
   devSessionStatusManager: DevSessionStatusManager
-  onAbort: () => Promise<void>
+  shopFqdn: string
+  onAbort?: () => Promise<void>
 }
 
 const DevSessionUI: FunctionComponent<DevSesionUIProps> = ({
   abortController,
   processes,
   devSessionStatusManager,
+  shopFqdn,
   onAbort,
 }) => {
   const {isRawModeSupported: canUseShortcuts} = useStdin()
 
   const [isShuttingDownMessage, setIsShuttingDownMessage] = useState<string | undefined>(undefined)
   const [error, setError] = useState<string | undefined>(undefined)
   const [status, setStatus] = useState<DevSessionStatus>(devSessionStatusManager.status)
+  const [persistentDevMessage, setPersistentDevMessage] = useState<string | undefined>(undefined)
 
   const {isAborted} = useAbortSignal(abortController.signal, async (err) => {
     if (err) {
       setIsShuttingDownMessage('Shutting down dev because of an error ...')
     } else {
-      setIsShuttingDownMessage('Shutting down dev ...')
-      setTimeout(() => {
-        if (isUnitTest()) return
-        treeKill(process.pid, 'SIGINT', false, () => {
-          process.exit(0)
-        })
-      }, 2000)
+      const appPreviewReady = devSessionStatusManager.status.isReady
+      if (isTruthy(process.env.SHOPIFY_CLI_PERSISTENT_DEV) && appPreviewReady) {
+        setPersistentDevMessage(
+          `Your app preview is still active in "${shopFqdn}". You can close it by running \`shopify app dev clean\`.`,
+        )
+      } else {
+        setIsShuttingDownMessage('Shutting down dev ...')
+        await onAbort?.()
```

Comment:
Why is `await onAbort?.()` only in this callpath (and not also with we are showing the persistent dev message)?

Reply:
onAbort basically calls the `devSessionDelete` mutation, we only want that in the non persistent case.